### PR TITLE
Update cmake template

### PR DESCRIPTION
**Pull request by jwillikers**
Reduce boilerplate and ensure that both the `Visual Studio` and `msvc` compilers are supported.

---

- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: docs/package_templates/cmake_package/all/conanfile.py [hunk 1]
```diff
@@ -89,8 +88,7 @@ def build_requirements(self):
         self.tool_requires("tool/x.y.z")
 
     def source(self):
-        get(self, **self.conan_data["sources"][self.version],
-                  destination=self.source_folder, strip_root=True)
```
Comment (prince-chrismc) [inline docs/package_templates/cmake_package/all/conanfile.py:93]: Where is the default destination? Does it get it from layout?

File: docs/package_templates/cmake_package/all/test_package/test_package.cpp [hunk 2]
```diff
@@ -4,11 +4,11 @@
 
 
 int main(void) {
-    std::cout << "Create a minimal usage for the target project here." << std::endl;
-    std::cout << "Avoid big examples, bigger than 100 lines" << std::endl;
-    std::cout << "Avoid networking connections." << std::endl;
-    std::cout << "Avoid background apps or servers." << std::endl;
-    std::cout << "The propose is testing the generated artifacts only." << std::endl;
+    std::cout << "Create a minimal usage for the target project here.\n";
```
Comment (prince-chrismc) [inline docs/package_templates/cmake_package/all/test_package/test_package.cpp:7]: I bet there's a meme for this diff 🤣

**Review by prince-chrismc — COMMENTED** [review]
One q https://github.com/conan-io/conan-center-index/pull/13237/files#r984717555

File: docs/package_templates/cmake_package/all/conanfile.py (hunk 1 above)
Comment (jwillikers) [reply docs/package_templates/cmake_package/all/conanfile.py:93]: It's just the current working directory by default. Documented [here](https://docs.conan.io/en/latest/reference/conanfile/tools/files/downloads.html#conan-tools-files-get). In the source method, the current working directory is set to `self.source_folder`.

File: docs/package_templates/cmake_package/all/test_package/test_package.cpp (hunk 2 above)
Comment (jwillikers) [reply docs/package_templates/cmake_package/all/test_package/test_package.cpp:7]: -_-

File: docs/package_templates/cmake_package/all/conanfile.py (hunk 1 above)
Comment (uilianries) [reply docs/package_templates/cmake_package/all/conanfile.py:93]: Revert, keep it to the source_folder. It's largely used in CCI and explicit makes it easier to understand the recipe.

File: docs/package_templates/cmake_package/all/test_package/test_package.cpp (hunk 2 above)
Comment (uilianries) [reply docs/package_templates/cmake_package/all/test_package/test_package.cpp:7]: The Bike Shed.

Actually, you coult convert it to a block comment I think. I totally forgot when I wrote that code.

File: docs/package_templates/cmake_package/all/conanfile.py [hunk 3]
```diff
@@ -77,9 +73,12 @@ def validate(self):
         # validate the minimum cpp standard supported. For C++ projects only
         if self.info.settings.compiler.cppstd:
             check_min_cppstd(self, self._minimum_cpp_standard)
-        minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
-        if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
-            raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._minimum_cpp_standard}, which your compiler does not support.")
+        if is_msvc(self):
+            check_min_vs(self, 191)
+        else:
```
Comment (uilianries) [inline docs/package_templates/cmake_package/all/conanfile.py:78]:
```suggestion
        check_min_vs(self, 191)
        if not is_msvc(self):
```
There is an advantage, `check_min_vs` only triggers its check it we are running msvc, internally, it checks it.

File: docs/package_templates/cmake_package/all/test_package/test_package.cpp (hunk 2 above)
Comment (jwillikers) [reply docs/package_templates/cmake_package/all/test_package/test_package.cpp]: I combined it to a single output statement.

File: docs/package_templates/cmake_package/all/conanfile.py (hunk 1 above)
Comment (jwillikers) [reply docs/package_templates/cmake_package/all/conanfile.py]: Done.

File: docs/package_templates/cmake_package/all/test_package/test_package.cpp [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
-    std::cout << "Create a minimal usage for the target project here." << std::endl;
-    std::cout << "Avoid big examples, bigger than 100 lines" << std::endl;
-    std::cout << "Avoid networking connections." << std::endl;
-    std::cout << "Avoid background apps or servers." << std::endl;
-    std::cout << "The propose is testing the generated artifacts only." << std::endl;
+  std::cout << "Create a minimal usage for the target project here.\n"
+               "Avoid big examples, bigger than 100 lines.\n"
+               "Avoid networking connections.\n"
+               "Avoid background apps or servers.\n"
+               "The propose is testing the generated artifacts only.\n";
```
Comment (chausner) [inline docs/package_templates/cmake_package/all/test_package/test_package.cpp:11]:
```suggestion
  std::cout << "Create a minimal usage sample for the target project here.\n"
               "Avoid big examples, longer than 100 lines.\n"
               "Avoid network connections.\n"
               "Avoid background apps or servers.\n"
               "The purpose is testing the generated artifacts only.\n";
```

Comment (jwillikers) [reply docs/package_templates/cmake_package/all/test_package/test_package.cpp]: Done.

**Review by prince-chrismc — APPROVED** [review]

File: docs/package_templates/cmake_package/all/conanfile.py [hunk 5]
```diff
@@ -58,12 +57,9 @@ def configure(self):
                 del self.options.fPIC # once removed by config_options, need try..except for a second del
             except Exception:
                 pass
-        try:
-            del self.settings.compiler.libcxx # for plain C projects only
-        except Exception:
-            pass
         try:
             del self.settings.compiler.cppstd # for plain C projects only
+            del self.settings.compiler.libcxx # for plain C projects only
```
Comment (uilianries) [inline docs/package_templates/cmake_package/all/conanfile.py:62]: Revert. We can't guarantee cppstd is evaluated.

File: docs/package_templates/cmake_package/all/test_package/test_package.cpp [hunk 6]
```diff
@@ -1,16 +1,16 @@
 #include <cstdlib>
 #include <iostream>
-#include "package/foobar.hpp"
 
+#include "package/foobar.hpp"
```
Comment (uilianries) [inline docs/package_templates/cmake_package/all/test_package/test_package.cpp:4]: bikeshedding, revert, please.

**Review by uilianries — CHANGES_REQUESTED** [review]
Keep only `check_min_vs`. On Conan 1.53.0 we will have safe delete for settings.compilers

File: docs/package_templates/cmake_package/all/conanfile.py (hunk 5 above)
Comment (jwillikers) [reply docs/package_templates/cmake_package/all/conanfile.py]: I gotcha, hadn't thought of that. Done.

File: docs/package_templates/cmake_package/all/test_package/test_package.cpp (hunk 6 above)
Comment (jwillikers) [reply docs/package_templates/cmake_package/all/test_package/test_package.cpp:3]: Done.

Comment (jwillikers) [reply docs/package_templates/cmake_package/all/test_package/test_package.cpp:3]: I reverted the line, you mean the file?

Comment (uilianries) [reply docs/package_templates/cmake_package/all/test_package/test_package.cpp:3]: yes

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM, thank you for this simplification! 



TODO: I think we should document, or at least, point a link with Visual Studio version vs msvc in docs.

**Pull request merged**